### DISCUSSION TITLE
Sound driver parameter names are not consistent

### DISCUSSION THREAD

**Discussion by krysclarke** [OpenTTD/OpenTTD#10106] [General]
As part of digging through the src/sound/allegro_s.cpp, I noticed that it looks for parameters of 'hz' and 'samples', yet the output of `openttd --help` doesn't mention 'samples' as a possible parameter to sound drivers.
Digging around a bit more in the other sound driver .cpp's, I found:
- allegro, sdl & sdl2 take 'hz' & 'samples'
- cocoa takes 'hz' ONLY
- xaudio2 & win32 take 'hz' & 'bufsize'

The relevant line of the output from `openttd --help`:
`  -s drv              = Set sound driver (see below) (param bufsize,hz)`

In all cases, if the parameter is missing, a sensible default is provided, but this inconsistency COULD cause confusion if someone wants to use the allegro, sdl or sdl2 drivers and isn't noticing any changes in-game when setting 'bufsize' on the command line.

My suggested fix is:
1. remove the `(param bufsize,hz)` text from the line shown above; and
2. list the supported parameters against each of the sound drivers in the `List of sound drivers:` section of the `--help` text.

**Comment by LordAro** [comment]
Go for it.

**Reply by krysclarke** [reply]
Step 1 is straight forward.
Step 2 is going to be a bit more involved if I want to avoid hard-coding entries
